fdbserver/Coordination.actor.cpp

```diff
@@ -562,46 +573,113 @@ ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf, OnDemandStore
 				info.forward = forward.get().serializedInfo;
 				req.reply.send(CachedSerialization<ClientDBInfo>(info));
 			} else {
-				regs.getInterface(req.clusterKey, id).openDatabase.send(req);
+				StringRef clusterName = ccf->getConnectionString().clusterKeyName();
+				if (! SERVER_KNOBS->ENABLE_CROSS_CLUSTER_SUPPORT &&
+					getClusterDescriptor(req.clusterKey).compare(clusterName)) {
+					TraceEvent(SevError, "CCFMismatch")
```

Comment:
Logging a `SevError` would cause this CD process to be marked as unhealthy in cluster status and requires operator to look into it, which in this case is incorrect. @ajbeamon to confirm.

Reply:
Yeah, this does seem awkward. The server process here has done nothing wrong and is also not in a bad state, so it probably shouldn't record a `SevError`. It would be reasonable for the process that sent the message to the coordinator to log an error.

Reply:
I'll change this to `SevWarn` and in the follow up PR modify callers to return a real error if they receive “wrong_connection_file”.